eng/common/templates/steps/source-build.yml

```diff
@@ -83,6 +83,10 @@ steps:
       assetManifestFileName=SourceBuild_${{ parameters.platform.name }}.xml
     fi
 
+    # Setup environment variables to support offline builds before calling into
+    # the build script.
+    export PUPPETEER_SKIP_DOWNLOAD=1
```

Comment:
This is not a robust solution - it doesn't get applied to the product source-build or when a dev builds aspnetcore in SB mode locally.  This should be set in the [SourceBuild.props](https://github.com/dotnet/aspnetcore/blob/main/eng/SourceBuild.props).  e.g. specify in `EnvironmentVariables=` in the npm build `exec`

Reply:
Fair enough, I'll update it